### PR TITLE
CAPBM machine manager updates

### PR DESCRIPTION
**Pull request by Jaakko-Os**
- v1alpha2 changes
- lint fixes
- unit tests will follow in a separate comit

### CODE REVIEW

File: baremetal/machine_manager.go [hunk 1]
```diff
@@ -133,98 +136,98 @@ func (mgr *MachineManager) GetProviderID(ctx context.Context) (*string, error) {
...
+func (m *MachineManager) Close() error {
+	return m.patchHelper.Patch(context.TODO(), m.BareMetalMachine)
 }
 
-// Create creates a machine and is invoked by the Machine Controller
-func (mgr *MachineManager) Associate(ctx context.Context) error {
-	mgr.Log.Info("Creating machine")
+// Associate associates a machine and is invoked by the Machine Controller
+func (m *MachineManager) Associate(ctx context.Context) error {
+	log.Printf("Associating machine %v .", m.Machine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: This is on a baremetal machine. Also please use m.Log.Info instead of Printf

File: baremetal/machine_manager.go [hunk 2]
```diff
@@ -133,98 +136,98 @@ func (mgr *MachineManager) GetProviderID(ctx context.Context) (*string, error) {
...
 		if err != nil {
 			return err
 		}
 		if host == nil {
-			mgr.Log.Info("No available host found. Requeuing.")
+			m.Log.Info("No available host found. Requeuing.")
 			return &RequeueAfterError{RequeueAfter: requeueAfter}
 		}
-		mgr.Log.Info("Associating machine with host", "host", host.Name)
+		log.Printf("Associating machine %s with host %s", m.Machine.Name, host.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 3]
```diff
@@ -133,98 +136,98 @@ func (mgr *MachineManager) GetProviderID(ctx context.Context) (*string, error) {
...
 	} else {
-		mgr.Log.Info("Machine already associated with host", "host", host.Name)
+		log.Printf("Machine %s already associated with host %s", m.Machine.Name, host.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 4]
```diff
@@ -247,56 +250,55 @@ func (mgr *MachineManager) mergeUserData(ctx context.Context) error {
...
 	}
 	return nil
 }
 
-// Delete deletes a machine and is invoked by the Machine Controller
-func (mgr *MachineManager) Delete(ctx context.Context) error {
-	mgr.Log.Info("Deleting machine")
+// Delete deletes a bare metal machine and is invoked by the Machine Controller
+func (m *MachineManager) Delete(ctx context.Context) error {
+	log.Printf("Deleting bare metal machine %v .", m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 5]
```diff
@@ -247,56 +250,55 @@ func (mgr *MachineManager) mergeUserData(ctx context.Context) error {
...
 
 	if host != nil && host.Spec.ConsumerRef != nil {
-		mgr.Log.Info("Updating host", "host", host.Name)
-		// don't remove the ConsumerRef if it references some other machine
-		if !consumerRefMatches(host.Spec.ConsumerRef, mgr.Machine) {
-			mgr.Log.Info("host associated with another machine",
-				"host", host.Name)
+		// don't remove the ConsumerRef if it references some other bare metal machine
+		if !consumerRefMatches(host.Spec.ConsumerRef, m.BareMetalMachine) {
+			log.Printf("host associated with %v, not bare metal machine %v.",
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 6]
```diff
@@ -319,73 +321,75 @@ func (mgr *MachineManager) Delete(ctx context.Context) error {
...
-	mgr.Log.Info("finished deleting machine")
+	m.Log.Info("finished deleting bare metal machine")
 	return nil
 }
 
 // Update updates a machine and is invoked by the Machine Controller
-func (mgr *MachineManager) Update(ctx context.Context) error {
-	mgr.Log.Info("Updating machine")
+func (m *MachineManager) Update(ctx context.Context) error {
+	log.Printf("Updating machine %v .", m.Machine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 7]
```diff
@@ -319,73 +321,75 @@ func (mgr *MachineManager) Delete(ctx context.Context) error {
...
 		return err
 	}
 
-	if err := mgr.updateMachineStatus(ctx, host); err != nil {
+	if err := m.updateMachineStatus(ctx, host); err != nil {
 		return err
 	}
 
-	mgr.Log.Info("Finished updating machine")
+	log.Printf("Finished updating machine %v .", m.Machine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 8]
```diff
@@ -319,73 +321,75 @@ func (mgr *MachineManager) Delete(ctx context.Context) error {
...
 	return nil
 }
 
-// Exists tests for the existence of a machine and is invoked by the Machine Controller
-func (mgr *MachineManager) Exists(ctx context.Context) (bool, error) {
-	mgr.Log.Info("Checking if machine exists.")
-	host, err := mgr.getHost(ctx)
+// Exists tests for the existence of a bare metal machine and is invoked by the Machine Controller
+func (m *MachineManager) Exists(ctx context.Context) (bool, error) {
+	log.Printf("Checking if bare metal machine %v exists.", m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 9]
```diff
@@ -319,73 +321,75 @@ func (mgr *MachineManager) Delete(ctx context.Context) error {
...
+	host, err := m.getHost(ctx)
 	if err != nil {
 		return false, err
 	}
 	if host == nil {
-		mgr.Log.Info("Machine does not exist.")
+		log.Printf("BareMetalMachine %v does not exist.", m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 10]
```diff
@@ -319,73 +321,75 @@ func (mgr *MachineManager) Delete(ctx context.Context) error {
...
 		return false, nil
 	}
-	mgr.Log.Info("Machine exists.")
+	log.Printf("BareMetalMachine %v exists.", m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 11]
```diff
@@ -394,22 +398,22 @@ func (mgr *MachineManager) Exists(ctx context.Context) (bool, error) {
 // (https://github.com/kubernetes-sigs/cluster-api/issues/160).
 
 // GetIP returns IP address of the machine in the cluster.
-func (mgr *MachineManager) GetIP() (string, error) {
-	mgr.Log.Info("Getting IP of machine")
+func (m *MachineManager) GetIP() (string, error) {
+	log.Printf("Getting IP of machine %v .", m.Machine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 12]
```diff
@@ -394,22 +398,22 @@ func (mgr *MachineManager) Exists(ctx context.Context) (bool, error) {
...
 	return "", fmt.Errorf("TODO: Not yet implemented")
 }
 
 // GetKubeConfig gets a kubeconfig from the running control plane.
-func (mgr *MachineManager) GetKubeConfig() (string, error) {
-	mgr.Log.Info("Getting Kubeconfig.")
+func (m *MachineManager) GetKubeConfig() (string, error) {
+	log.Printf("Getting IP of machine %v .", m.Machine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf and wrong comment in printf

File: baremetal/machine_manager.go [hunk 13]
```diff
@@ -458,23 +462,22 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
 	// I think it's because we have a local cache of all BareMetalHosts.
 	labelSelector := labels.NewSelector()
 	var reqs labels.Requirements
-	for labelKey, labelVal := range mgr.BareMetalMachine.Spec.HostSelector.MatchLabels {
-		mgr.Log.Info("Adding requirement to match label", "label key", labelKey,
-			"label value", labelVal)
+
+	for labelKey, labelVal := range m.BareMetalMachine.Spec.HostSelector.MatchLabels {
+		log.Printf("Adding requirement to match label: '%s' == '%s'", labelKey, labelVal)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 14]
```diff
@@ -458,23 +462,22 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
...
+			m.Log.Error(err, "Failed to create MatchLabel requirement, not choosing host")
 			return nil, err
 		}
 		reqs = append(reqs, *r)
 	}
-	for _, req := range mgr.BareMetalMachine.Spec.HostSelector.MatchExpressions {
-		mgr.Log.Info("Adding requirement to match label", "label key", req.Key,
-			"label operator", req.Operator, "label value", req.Values)
+	for _, req := range m.BareMetalMachine.Spec.HostSelector.MatchExpressions {
+		log.Printf("Adding requirement to match label: '%s' %s '%s'", req.Key, req.Operator, req.Values)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 15]
```diff
@@ -486,17 +489,17 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
 	for i, host := range hosts.Items {
 		if host.Available() {
 			if labelSelector.Matches(labels.Set(host.ObjectMeta.Labels)) {
-				mgr.Log.Info("Host matched hostSelector for Machine", "host", host.Name)
+				log.Printf("Host '%s' matched hostSelector for BareMetalMachine '%s'", host.Name, m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 16]
```diff
@@ -486,17 +489,17 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
...
 				availableHosts = append(availableHosts, &hosts.Items[i])
 			} else {
-				mgr.Log.Info("Host did not match hostSelector for Machine", "host", host.Name)
+				log.Printf("Host '%s' did not match hostSelector for BareMetalMachine '%s'", host.Name, m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 17]
```diff
@@ -486,17 +489,17 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
...
 			}
-		} else if host.Spec.ConsumerRef != nil && consumerRefMatches(host.Spec.ConsumerRef, mgr.Machine) {
-			mgr.Log.Info("found host with existing ConsumerRef", "host", host.Name)
+		} else if host.Spec.ConsumerRef != nil && consumerRefMatches(host.Spec.ConsumerRef, m.BareMetalMachine) {
+			log.Printf("found host %s with existing ConsumerRef", host.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 18]
```diff
@@ -486,17 +489,17 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
...
 			return &hosts.Items[i], nil
 		}
 	}
-	mgr.Log.Info(fmt.Sprintf("%d hosts available while choosing host for machine", len(availableHosts)))
+	log.Printf("%d hosts available while choosing host for bare metal machine '%s'", len(availableHosts), m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go [hunk 19]
```diff
@@ -509,18 +512,18 @@ func (mgr *MachineManager) chooseHost(ctx context.Context) (*bmh.BareMetalHost,
...
 	}
-	if consumer.Kind != machine.Kind {
+	if consumer.Kind != bmmachine.Kind {
 		return false
 	}
-	if consumer.APIVersion != machine.APIVersion {
+	if consumer.APIVersion != bmmachine.APIVersion {
 		return false
 	}
 	return true
```
Comment (maelk) [inline baremetal/machine_manager.go]: Please leave this change out, and do it in another commit, it breaks more things than it seems

File: baremetal/machine_manager.go [hunk 20]
```diff
@@ -538,57 +541,57 @@ func (mgr *MachineManager) setHostSpec(ctx context.Context, host *bmh.BareMetalH
...
+		m.Log.Error(err, "Error parsing annotation value", "annotation key", hostKey)
 		return err
 	}
 	existing, ok := annotations[HostAnnotation]
 	if ok {
 		if existing == hostKey {
 			return nil
 		}
-		mgr.Log.Info("Warning: found stray annotation for host on machine. Overwriting.", "host", existing)
+		log.Printf("Warning: found stray annotation for host %s on machine %s. Overwriting.", existing, m.BareMetalMachine.Name)
```
Comment (maelk) [inline baremetal/machine_manager.go]: m.Log.Info instead of log.Printf

File: baremetal/machine_manager.go (hunk 1 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 2 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 3 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 4 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 5 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 6 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 7 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 8 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 9 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 10 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 11 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 12 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 13 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 14 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 15 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 16 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 17 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 18 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done

File: baremetal/machine_manager.go (hunk 19 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: reverted back to machine

File: baremetal/machine_manager.go (hunk 20 above)
Comment (Jaakko-Os) [reply baremetal/machine_manager.go]: done
